slurm/src/slurmcc/cli.py

```diff
@@ -731,13 +731,23 @@ def _generate_gres_conf(partitions: List[partitionlib.Partition], writer: TextIO
             # cut out 1gb so that the node reports at least this amount of memory. - recommended by schedmd
 
             if partition.gpu_count:
+                # TODO: azure sku should eventually indicate the GPU type
+                try:
+                    has_amd_gpu = partition.gpu_family.lower() == "amd"
+                except AttributeError:   
+                    has_amd_gpu = "mi300" in partition.machine_type.lower()
+
                 if partition.gpu_count > 1:
-                    nvidia_devices = "/dev/nvidia[0-{}]".format(partition.gpu_count - 1)
+                    nvidia_devices = "/dev/nvidia[0-{}]".format(partition.gpu_count - 1)                    
```

Comment:
Is it possible to move this into a "generate_nvidia_gpu_device()" function of some sort and then  just give it number of devices to generate? same for amd devices.